### DISCUSSION TITLE
Communicate with PLM on Different Host

### DISCUSSION THREAD

**Discussion by CodeOnCommand** [TD22057/insteon-mqtt#510] [Ideas]
Would be interesting to control Insteon devices via Home Assistant UI that are currently controlled via openHAB w/PLM on a different host on the local network.

**Comment by EnGamma** [comment]
I've wanted to do something similar: basically have devices respond to and
report to 2 (or more) different PLM's on different platforms.   I think
this is certainly possible, but unfortunately insteon-mqtt does not
tolerate well the second PLM.   There is no provision to acknowledge its
existence in the config file and therefore whenever a device/modem sync is
done it wants to (and does) remove links in the devices to the second PLM.
 This is on my list of desired improvements to insteon-mqtt--to at least
tolerate the existence of other PLMs in the home.   I think it would be
fairly straightforward config file and software change, but I haven't had
time yet to look into it.

**Reply by CodeOnCommand** [reply]
I should have been a little clearer in what I was suggesting.

Using the following as an example:

Home Assistant is running in a Docker in a VM on a Linux Host with NO PLM attached.

openHAB is running on a Raspberry Pi with a single PLM attached.

It would be very neat to be able to use insteon-mqtt to control the PLM attached to openHAB and keep the UI of both openHAB and Home Assistant in sync.

So during the configuration of insteon-mqtt within Home Assistant a user would enter the IP address and the associated USB device path of the remote PLM. 

USB over IP is already a thing something similar integrated withing insteon-mqtt would be killer.

**Reply by krkeegan** [reply]
Just connect insteon-mqtt to the PLM. Then setup both home assistant and openhab to interact with the insteon devices over MQTT. Everything will stay in sync.

**Reply by CodeOnCommand** [reply]
That is an idea, was under the impression that insteon-mqtt was for Home Assistant.   I am hesitant to tinker with my openHAB setup too much as minor changes seem to cause unexpected hiccups especially with Hue Emulation and Alexa. 

Thought I found a simple solution to my situation when I stumbled across insteon-mqtt. 

As you are probably aware openHAB deals with devices via Things, Items and Channels and then there are MQTT Things which also has Channels.   Restoring to scripts to publish state when an Item has changed meaning the state topic in Mosquitto is not actually supplied by the the device which is what I thought was to happen......but I guess that is for a different post on a different forum.

Feel free to post a guide or two on Insteon and MQTT on openHAB  😄 

Thank you for your efforts and keep up the good work.

**Reply by krkeegan** [reply]
I haven't ever used openHAB, but Insteon-MQTT is largely platform agnostic.  We do have a docker container that can be installed as an addon in HomeAssistant, but it is pretty basic.

Insteon-MQTT can be run from the command line as a python script.  Once up and running, you can interact with it through MQTT or from the command line.

**Reply by kpfleming** [reply]
The only part of Insteon-MQTT that is Home Assistant specific is the implementation of the discovery protocol, which is optional. The rest would be easy to support in any system that can use MQTT.
